From b6d025ba399504b7af2ebeeb3216c357fbea73e5 Mon Sep 17 00:00:00 2001
From: Anoob Soman <anoob.soman@citrix.com>
Date: Tue, 15 Aug 2017 15:46:12 +0100
Subject: [PATCH] xen_pt: Allow PCI_CLASS_DISPLAY_OTHER igd devices to be
 passthrued

On Broadwell machines, igd PCI devices comes up with class_code set
to PCI_CLASS_DISPLAY_OTHER. Allow them to be pass-throughed.
diff --git a/hw/xen/xen_pt.h b/hw/xen/xen_pt.h
index d422482..1f043b4 100644
--- a/hw/xen/xen_pt.h
+++ b/hw/xen/xen_pt.h
@@ -337,7 +337,8 @@ extern bool has_igd_gfx_passthru;
 static inline bool is_igd_vga_passthrough(XenHostPCIDevice *dev)
 {
     return (has_igd_gfx_passthru
-            && ((dev->class_code >> 0x8) == PCI_CLASS_DISPLAY_VGA));
+            && (((dev->class_code >> 0x8) == PCI_CLASS_DISPLAY_VGA)
+            || ((dev->class_code >> 0x8) == PCI_CLASS_DISPLAY_OTHER)));
 }
 int xen_pt_register_vga_regions(XenPCIPassthroughState *s);
 int xen_pt_unregister_vga_regions(XenPCIPassthroughState *s);
